### PR TITLE
Validate the model name

### LINKED ISSUE #2055: No user feedback on model name verification
I entered an invalid model name. No error.

Deploy button was non-responsive.  Checked JS console and found:

```
Uncaught failed to create config: creating config from values failed: "my first model" is not a valid name: model names may only contain lowercase letters, digits and hyphens.
```

That error needs to be caught and shown to the user.

### PR DESCRIPTION
**Pull request by huwshimi**
Fixes #2055.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5916/
Test PASSed.

**Comment by huwshimi** [comment]
QA:
- add a service
- open the deployment flow
- type names into the model name input, it should warn you if you type anything but lowercase, numbers and hyphens

File: jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js [hunk 1]
```diff
@@ -554,6 +554,11 @@ YUI.add('deployment-flow', function() {
               validate={[{
                 regex: /\S+/,
                 error: 'This field is required.'
+              }, {
+                regex: /^[a-z0-9]([a-z0-9.-]*[a-z0-9])?$/,
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js]: Maybe add a commenti pointing to where you took this regexp?
I'd assume https://github.com/juju/names/blob/master/model.go#L20

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**